model/map/src/main/java/org/keycloak/models/map/user/MapUserProviderFactory.java

```diff
@@ -40,6 +40,8 @@
  */
 public class MapUserProviderFactory extends AbstractMapProviderFactory<MapUserProvider, MapUserEntity, UserModel> implements UserProviderFactory<MapUserProvider>, InvalidationHandler {
 
+    public static final String USERNAME_CASE_SENSITIVE = "username-case-sensitive";
```

Comment:
The attribute name should be clearly keycloak internal, e.g.:
```suggestion
    public static final String REALM_ATTR_USERNAME_CASE_SENSITIVE = "keycloak.username-search.case-sensitive";
```